homeassistant/components/trace/websocket_api.py

```diff
@@ -57,33 +57,47 @@ def async_setup(hass: HomeAssistant) -> None:
     }
 )
 def websocket_trace_get(hass, connection, msg):
-    """Get an automation or script trace."""
+    """Get an script or automation trace."""
     key = (msg["domain"], msg["item_id"])
     run_id = msg["run_id"]
 
-    trace = get_debug_trace(hass, key, run_id)
+    trace = hass.data[DATA_TRACE][key][run_id]
```

Comment:
You should catch `KeyError` and return a NOT_FOUND error.

Reply:
Let's fix that in a follow-up PR.

Reply:
Fixed in https://github.com/home-assistant/core/pull/48502